File: regexlib/javascript_test_file/regexlib_1167.js
/* 1167
 * ^(([a-zA-Z0-9]+([\-])?[a-zA-Z0-9]+)+(\.)?)+[a-zA-Z]{2,6}$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"000"*8+"!1 __EOA(i or ii)"
 */
var REGEX = /^(([a-zA-Z0-9]+([\-])?[a-zA-Z0-9]+)+(\.)?)+[a-zA-Z]{2,6}$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '000'.repeat(i*1) + '!1 __EOA(i or ii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re2_test_file/regexlib_1167.py
# 1167
# ^(([a-zA-Z0-9]+([\-])?[a-zA-Z0-9]+)+(\.)?)+[a-zA-Z]{2,6}$
# EXPONENT
# nums:5
# EXPONENT AttackString:""+"000"*8+"!1 __EOA(i or ii)"

import re2 as re
from time import perf_counter

regex = """^(([a-zA-Z0-9]+([\-])?[a-zA-Z0-9]+)+(\.)?)+[a-zA-Z]{2,6}$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "000" * i * 1 + "!1 __EOA(i or ii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")